cirq/experiments/xeb_fitting.py

```diff
@@ -453,22 +459,36 @@ def _fit_exponential_decay(cycle_depths: np.ndarray, fidelities: np.ndarray) ->
         a: The first fit parameter that scales the exponential function, perhaps accounting for
             state prep and measurement (SPAM) error.
         layer_fid: The second fit parameters which serves as the base of the exponential.
+        a_std: The standard deviation of the `a` parameter estimate.
+        layer_fid_std: The standard deviation of the `layer_fid` parameter estimate.
     """
     cycle_depths = np.asarray(cycle_depths)
     fidelities = np.asarray(fidelities)
 
     # Get initial guess by linear least squares with logarithm of model
     positives = fidelities > 0
+    if np.sum(positives) <= 1:
+        return 0, 0, np.inf, np.inf
     cycle_depths_pos = cycle_depths[positives]
     log_fidelities = np.log(fidelities[positives])
     slope, intercept, _, _, _ = scipy.stats.linregress(cycle_depths_pos, log_fidelities)
     layer_fid_0 = np.clip(np.exp(slope), 0, 1)
     a_0 = np.clip(np.exp(intercept), 0, 1)
 
-    (a, layer_fid), _ = scipy.optimize.curve_fit(
-        exponential_decay, cycle_depths, fidelities, p0=(a_0, layer_fid_0), bounds=((0, 0), (1, 1))
-    )
-    return a, layer_fid
+    try:
+        (a, layer_fid), pcov = scipy.optimize.curve_fit(
```

Comment:
Is it possible for this to optimization to fail (not converge?)

Reply:
yes, that's why the try except is being added. I couldn't figure out how to make a test dataset for a unit test to trigger the behavior, but the real device sometimes does :)